src/sources/prometheus/parser.rs

```diff
@@ -1,301 +1,44 @@
-
-            current_header = match current_header.kind {
-                ParserType::Histogram => {
-                    // check if this is still a histogram
-                    if metric.name == format!("{}_bucket", current_header.name)
-                        || metric.name == format!("{}_count", current_header.name)
-                        || metric.name == format!("{}_sum", current_header.name)
-                    {
-                        current_header
-                    } else {
-                        // nope it's a new unrelated metric
-                        ParserHeader {
-                            name: metric.name,
-                            kind: ParserType::Untyped,
-                        }
-                    }
-                }
-                ParserType::Summary => {
-                    // check if this is still a summary
-                    if metric.name == current_header.name
-                        || metric.name == format!("{}_count", current_header.name)
-                        || metric.name == format!("{}_sum", current_header.name)
-                    {
-                        current_header
-                    } else {
-                        ParserHeader {
-                            name: metric.name,
-                            kind: ParserType::Untyped,
-                        }
-                    }
-                }
-                _ => {
-                    if metric.name == current_header.name {
-                        current_header
-                    } else {
-                        ParserHeader {
-                            name: metric.name,
-                            kind: ParserType::Untyped,
-                        }
-                    }
-                }
-            };
-
-            if let Some(lines) = result.get_mut(&current_header) {
-                lines.push(line);
-            } else {
-                result.insert(current_header.clone(), vec![line]);
-            }
-        }
-    }
-
-    Ok(result)
 }
 
 pub fn parse(packet: &str) -> Result<Vec<Metric>, ParserError> {
     let mut result = Vec::new();
 
-    for (header, group) in group_metrics(packet)? {
+    for group in prometheus_parser::group_metrics(packet)? {
         // just a header without measurements
-        if group.is_empty() {
+        if group.metrics.is_empty() {
             continue;
         }
 
-        match header.kind {
-            ParserType::Counter => {
-                for line in group {
-                    let metric = parse_metric(&line)?;
-                    let tags = if !metric.tags.is_empty() {
-                        Some(metric.tags)
+        match group.metrics {
+            GroupKind::Counter(vec) => {
+                for metric in vec {
+                    let tags = if !metric.labels.is_empty() {
+                        Some(metric.labels)
                     } else {
                         None
                     };
```

Comment:
This pattern is repeated several times here and could use a helper to clean that up.